asf_heat_pump_suitability/pipeline/suitability/calculate_suitability_v2.py

```diff
@@ -0,0 +1,554 @@
+
+
+def score_dict_off_gas(
+    tech_type,
+    off_gas,
+    # solar
+):
+    """
+    Is it off-gas or does it have solar panels?
+    """
+    # TODO: add solar
+    if off_gas:
+        return offgas_scores.get(tech_type)
+    else:
+        return None
+
+
+def score_dict_property_density(tech_type, lsoa):
+    """
+    Is there a high property density in this LSOA? (% TBC)
+    """
+    property_density_per_lsoa = {22: 0.8, 55: 0.2}  # To define
+    threshold = 0.4
+    if property_density_per_lsoa.get(lsoa) > threshold:
+        return property_density_scores.get(tech_type)
+    else:
+        return None
+
+
+def score_dict_high_heat_demand(tech_type, ruc_two_fold):
+    """
+    Is this property in an urban LSOA/high heat demand density LSOA?
+    """
+    if ruc_two_fold == "Urban":
+        return high_heat_demand_scores.get(tech_type)
+    else:
+        return None
+
+
+def score_dict_anchor_properties(tech_type, lsoa):
+    """
+    Is this property in a LSOA with schools/hospitals/community centres/housing developments?
+    """
+    anchors_per_lsoa = {22: True, 55: False}  # To define
+    if anchors_per_lsoa.get(lsoa):
+        return anchor_properties_scores.get(tech_type)
+    else:
+        return None
+
+
+def get_property_scores(
+    ruc_two_fold: str,
+    off_gas: bool,
+    property_density: float,
+    outdoor_space: float,
+    listed_building: bool,
+    building_conservation_area: bool,
+    property_type: str,
+    epc_rating: str,
+) -> dict:
+    """
+    Calculate heat pump suitability score per tech type for the standard (S) and Nesta (N) views for an EPC property.
+    Tech types include: air-source heat pump (ASHP), ground-source heat pump (GSHP), shared ground loop (SGL), heat
+    network (HN).
+
+    Args:
+        ruc_two_fold (str):
+        off_gas (bool):
+        property_density (float):
+        outdoor_space (float):
+        listed_building (bool):
+        building_conservation_area (bool):
+        property_type (str):
+        epc_rating (str):
+
+    Returns:
+        dict: heat pump suitability scores per tech type.
+    """
+    all_scores = defaultdict(int)
+    for scores in [
+        score_dict_high_heat_demand(ruc_two_fold),
+        score_dict_off_gas(off_gas),
+        score_dict_property_density(property_density),
+        score_dict_garden_size(outdoor_space),
+        score_dict_external_space(outdoor_space),
+        score_dict_site_regs(listed_building, building_conservation_area),
+        score_dict_property_not_flat(property_type),
+        score_dict_epc_rating(epc_rating),
+    ]:
+        if scores:
+            for k, v in scores.items():
+                all_scores[k] += v
+
+    if not all_scores:
+        all_scores = {
+            "ASHP_S": 0,
+            "ASHP_N": 0,
+            "GSHP_S": 0,
+            "GSHP_N": 0,
+            "SGL_S": 0,
+            "SGL_N": 0,
+            "HN_S": 0,
+            "HN_N": 0,
+        }
+    return all_scores
+
+
```

Comment:
Refactored logic is from line 366 - 507 (3 functions). It should apply the same logic as before with a couple of minor differences:
- User specifies tech type and it calculates scores for 1 tech type at a time
- Scores for listed building status and conservation area status are calculated separately.